JavaScriptTasks/Session 2 - CoreJs(26-02-2020)/PartitionOn.JS

```diff
@@ -0,0 +1,70 @@
+/*PROBLEM STATEMENT :Partition On
+
+http://www.codewars.com/kata/partition-on */
+
+SOLUTION:
+
+function partitionOn(pred, items) {
+ var first=[],second=[];
+ var arrayLength=items.length;
+ var resultIndex=0,counter1=0,counter2=0;
+ for(var i=0;i<arrayLength;i++)
+ {
+ if(pred(items[i]))
+ {
+second[counter2]=items[i];
+counter2++;
+ }else{
+    first[counter1]=items[i];
+    counter1++;
+resultIndex=counter1;
+ }
+ }
+ for(var j=0;j<counter2;j++)
+ {
+    first[counter1]=second[j];
+    counter1++;
+ }
+
+ for(var k=0;k<arrayLength;k++)
+ {
+ items[k]=first[k];
+ }
+
+
+ return resultIndex;
+ 
+
+}
+
+
+
+
+
+
+//Aletnative Solution
+
+
+
+function partitionOn(pred, items) {
+ var len=items.length;
```

Comment:
Check with Krushna Dongre about this method logic

Reply:
modified the code as per reference